04 Strategy Library/88 52-Weeks High Effect/02 Method.html

```diff
@@ -0,0 +1,99 @@
+  More available objects for use in QuantConnect Fine Universe filter can be found <a href="https://www.quantconnect.com/docs/data-library/fundamentals">here</a>.
+<p>
+<div class="section-example-container">
+    <pre class="python">
+        self.BasicMaterialsIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.BasicMaterials]
+        self.ConsumerCyclicalIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.ConsumerCyclical]
+        self.FinancialServicesIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.FinancialServices]
+        self.RealEstateIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.RealEstate]
+        self.ConsumerDefensiveIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.ConsumerDefensive]
+        self.HealthcareIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.Healthcare]
+        self.UtilitiesIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.Utilities]
+        self.CommunicationServicesIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.CommunicationServices]
+        self.EnergyIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.Energy]
+        self.IndustrialsIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.Industrials]
+        self.TechnologyIndustry = [x for x in fine if x.AssetClassification.MorningstarSectorCode == MorningstarSectorCode.Technology]
+        
+    </pre>
+</div>
+<p>
+  <h4>Calculation of market capitalization</h4>
+  In FineSelectionFunction, we calculate the market capitalization for each stock in our universe.
+</p>
+  <div class="section-example-container">
+      <pre class="python">
+        hist = self.History(self.symbols, 252, Resolution.Daily)
+        for x in fineFiltered:
+            if str(x.Symbol) in hist.index.get_level_values(0):   #If the historical data is available
+                self.mktCap[x.Symbol] = float(x.EarningReports.BasicAverageShares.ThreeMonths) * hist.loc[str(x.Symbol)]['close'][-1]
+            else:
+                del self.symbols[self.symbols.index(x.Symbol)]    # Delete if we can't find any historical data
+                for industry in self.industry_list:
+                    if x in industry: del industry[industry.index(x)]
+      </pre>
+  </div>
+
+  <p>
+      <h4>Rolling window of stock prices</h4>
+      We also initialize rolling windows of 252 days for each stock in FineSelectionFunction so that we can use <code>max(self.windows[symbol])</code> to track its 52-weeks high.
```

Comment:
We also initialize _252-day Rolling Windows_ for each stock in FineSelectionFunction